SpatialGDK/Source/SpatialGDKTests/SpatialGDKServices/LocalDeploymentManager/LocalDeploymentManagerTest.cpp

```diff
@@ -122,6 +122,17 @@ DEFINE_LATENT_COMMAND_TWO_PARAMETERS(WaitForDeployment, FAutomationTestBase*, Te
 	const double NewTime = FPlatformTime::Seconds();
 	if (NewTime - StartTime >= MAX_WAIT_TIME_FOR_LOCAL_DEPLOYMENT_OPERATION)
 	{
+		// the given time for the deployment to start/stop has expired - test its current state
+		FLocalDeploymentManager* LocalDeploymentManager = GetLocalDeploymentManager();
+
+		if (ExpectedDeploymentState == EDeploymentState::IsRunning)
+		{
+			Test->TestTrue(TEXT("Deployment is running"), LocalDeploymentManager->IsLocalDeploymentRunning() && !LocalDeploymentManager->IsDeploymentStopping());
+		}
+		else
+		{
+			Test->TestFalse(TEXT("Deployment is not running"), LocalDeploymentManager->IsLocalDeploymentRunning() || LocalDeploymentManager->IsDeploymentStopping());
+		}
```

Comment:
I'm curious about your stances on adding the possibility that the `GIVEN` conditions in tests fail. I believe this is appropriate. Prior to this change, the `GIVEN_deployment_running_WHEN_deployment_stopped_THEN_deployment_not_running` test would be successful if a deployment is never started during setup of the test (`GIVEN` stage).

Reply:
I agree, so I've created a Jira ticket to address that https://improbableio.atlassian.net/browse/UNR-2331